[quant] Expose register activation post process hook function to user

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #42343 [quant][fix] Remove activation_post_process in qat modules
* **#42342 [quant] Expose register activation post process hook function to user**

Summary:
This is used in https://github.com/pytorch/pytorch/pull/42343
Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856711](https://our.internmc.facebook.com/intern/diff/D22856711)

jerryzh168 committed:
[quant] Expose register activation post process hook function to user

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/quantization/quantize.py:
```diff
@@ -80,6 +80,11 @@ def _observer_forward_pre_hook(self, input):
     # Returning nothing is Ok, Module._call_impl will intrepret this
     # as the pre_hook making no changes to the input, as desired
 
+def register_activation_post_process_hook(module):
```
Can you explain on why this refactoring is needed?

jerryzh168 replied:
We need to use this function in module definition, see next PR

raghuramank100 replied:
Sorry, didnt see that before, its clear now.

jerryzh168 committed:
Update on "[quant] Expose register activation post process hook function to user"


Summary:
This is used in https://github.com/pytorch/pytorch/pull/42343
Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22856711](https://our.internmc.facebook.com/intern/diff/D22856711)

[ghstack-poisoned]
